### PR TITLE
fix: add guest stars to ITraktShowCastAndCrew

### LINKED ISSUE #184: Adding GuestStars as ExtendedInfo when calling GetEpisodePeopleAsync does not return the guest starts
Postman Get  https://api.trakt.tv/shows/perry-mason-2020/seasons/1/episodes/8/people/?extended=guest_stars  returns many (36?) guest stars. However, the following code does not return any guest stars:

        private static async Task BugReport(string mySource, string myShow, uint mySeason = 0, uint myEpisode = 0)
        {
            TraktResponse<ITraktCastAndCrew> myPeopleEpisode;
            TraktExtendedInfo myExtendedInfo = new TraktExtendedInfo();

            myExtendedInfo.SetGuestStars();

            mySource = "episode";
            myShow = "perry-mason-2020";
            mySeason = 1;
            myEpisode = 8;

            myPeopleEpisode = await client.Episodes.GetEpisodePeopleAsync(myShow, mySeason, myEpisode, myExtendedInfo);
            if (myPeopleEpisode)
            {
                foreach (ITraktCastMember item in myPeopleEpisode.Value.Cast)
                {
                    Console.WriteLine(item.Character + " is played by " + item.Person.Name);
                }
            }
        }

Occurs in Trakt.Net version 1.0.2 under Windows 10 Visual Studio Community 2019 version 16.7.2

### PR DESCRIPTION
**Pull request by skyfrk**
Resolves #184.

### CODE REVIEW

**Comment by skyfrk** [comment]
Ready for review. Tell me if I should add more unit tests or how to aute-generate them :D

**Comment by henrikfroehling** [comment]
@skyfrk
Please target the branch `release-1.1.0`.

File: Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
+﻿using FluentAssertions;
```
Comment (henrikfroehling) [inline Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs:1]: Usings should be inside the namespace, like this:
```csharp
namespace TraktNet.Objects.Get.Tests.Shows.Json.Reader
{
    using FluentAssertions;
    using System.Linq;
    using System.Threading.Tasks;
    using Trakt.NET.Tests.Utility.Traits;
    using TraktNet.Objects.Get.Shows.Json.Reader;
    using Xunit;
```

It doesn't really matter. Just for consistency with the rest of the codebase.

File: Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using TraktNet.Objects.Get.Shows.Json.Reader;
+using Xunit;
+
+namespace TraktNet.Objects.Get.Tests.Shows.Json.Reader
+{
+    [Category("Objects.Get.Shows.JsonReader")]
+    public partial class ShowCastAndCrewObjectJsonReader_Tests
+    {
+        [Fact]
+        public async Task Test_ShowAliasObjectJsonReader_ReadObject_From_Json_String_Complete()
```
Comment (henrikfroehling) [inline Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs:14]:
```suggestion
        public async Task Test_ShowCastAndCrewObjectJsonReader_ReadObject_From_Json_String_Complete()
```

Comment (skyfrk) [reply Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs:14]: Oops

File: Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            var jsonReader = new ShowCastAndCrewObjectJsonReader();
+
+            var traktShowCastAndCrew = await jsonReader.ReadObjectAsync(JSON_COMPLETE);
+
+            traktShowCastAndCrew.Cast.First().Characters.First().Should().Be("Daenerys Targaryen");
```
Comment (henrikfroehling) [inline Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs:20]: Please don't use LINQ (`First()`).
Here you can write `traktShowCastAndCrew.Cast[0].Characters...`

File: Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Reader_Tests.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
+﻿using FluentAssertions;
```
Comment (henrikfroehling) [inline Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Reader_Tests.cs:1]: Usings inside the namespace...

File: Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Reader_Tests.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        public async Task Test_ShowCastAndCrewObjectJsonReader_ReadObject_From_JsonReader_Complete()
+        {
+            var traktJsonReader = new ShowCastAndCrewObjectJsonReader();
+
+            using (var reader = new StringReader(JSON_COMPLETE))
+            using (var jsonReader = new JsonTextReader(reader))
+            {
+                var traktShowCastAndCrew = await traktJsonReader.ReadObjectAsync(jsonReader);
+
+                traktShowCastAndCrew.Cast.First().Characters.First().Should().Be("Daenerys Targaryen");
```
Comment (henrikfroehling) [inline Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Reader_Tests.cs:25]: Avoid LINQ-queries and directly use the index operator.

File: Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Stream_Tests.cs [hunk 6]
```diff
@@ -0,0 +1,78 @@
+﻿using FluentAssertions;
```
Comment (henrikfroehling) [inline Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Stream_Tests.cs:1]: Usings inside the namespace...

File: Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Stream_Tests.cs [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+        public async Task Test_ShowCastAndCrewObjectJsonReader_ReadObject_From_Stream_Complete()
+        {
+
+            var traktJsonReader = new ShowCastAndCrewObjectJsonReader();
+
+            using (var stream = JSON_COMPLETE.ToStream())
+            {
+                var traktShowCastAndCrew = await traktJsonReader.ReadObjectAsync(stream);
+
+                traktShowCastAndCrew.Cast.First().Characters.First().Should().Be("Daenerys Targaryen");
```
Comment (henrikfroehling) [inline Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Stream_Tests.cs:26]: Avoid LINQ-queries and directly use the index operator.

File: Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Stream_Tests.cs [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+namespace TraktNet.Objects.Get.Tests.Shows.Json.Reader
+{
+    [Category("Objects.Get.Shows.JsonReader")]
+    public partial class ShowCastAndCrewObjectJsonReader_Tests
+    {
+        [Fact]
+        public async Task Test_ShowCastAndCrewObjectJsonReader_ReadObject_From_Stream_Complete()
+        {
+
```
Comment (henrikfroehling) [inline Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Stream_Tests.cs:19]: Unnecessary empty lines.

**Comment by henrikfroehling** [comment]
> Ready for review. Tell me if I should add more unit tests or how to auto-generate them :D

Unit tests are enough. Auto-generating them is copy-paste and search-and-replace. :-D

File: Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs (hunk 3 above)
Comment (skyfrk) [reply Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs:20]: image

Am I missing something here?

Comment (henrikfroehling) [reply Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs:20]: Ah, okay. Yes, doesn't work without minimum LINQ. :-D

```csharp
ITraktShowCastMember[] cast = traktShowCastAndCrew.Cast.ToArray();
cast[0].Characters.First().Should().Be("Daenerys Targaryen");
```

Comment (skyfrk) [reply Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs:20]: So should I convert the collections and then assert them or leave it the way it is?

Comment (henrikfroehling) [reply Source/Tests/Trakt.NET.Objects.Get.Tests/Shows/Json/Reader/ShowCastAndCrewObjectJsonReader/ShowCastAndCrewObjectJsonReader_Json_String_Tests.cs:20]: You can do it like in this file:
https://github.com/henrikfroehling/Trakt.NET/blob/a198696d11542c5235bcbee2dc42e1052774bda1/Source/Tests/Trakt.NET.Objects.Get.Tests/People/Credits/Json/Reader/PersonMovieCreditsObjectJsonReader/PersonMovieCreditsObjectJsonReader_Json_String_Tests.cs#L24

**Comment by skyfrk** [comment]
Ready for review again!

**Review by henrikfroehling — APPROVED** [review]

**Comment by henrikfroehling** [comment]
Thanks for the contribution. 👍🏻

**Pull request merged**